src/scss/components/_button.scss

```diff
@@ -53,97 +53,186 @@ $btn-font-family: $font-family-headings;
     background-color: $btn-disabled-bg;
     color: $btn-disabled-color;
     pointer-events: none;
+    border-color: $btn-disabled-bg;
   }
 }
 
-
 @each $color, $value in $theme-colors {
-  .btn-#{$color} {
-    color: color-yiq($value) !important;
-    text-decoration: none;
+  .btn-#{$color}, .btn-outline-#{$color} {
+    min-height: 40px;
+    color: color-yiq($value);
+    text-decoration: $link-decoration;
 
     @include hover() {
       background-color: theme-color-level($color, 1);
       border-color: theme-color-level($color, 1);
       box-shadow: 0 0 0 $btn-hover-width theme-color-level($color, 1);
     }
 
+    &:visited{
+      color: color-yiq($value);
+    }
+
     &:focus,
     &.focus {
       box-shadow: 0 0 0 $input-btn-focus-width $box-shadow-button;
-      text-decoration: none;
+      text-decoration: $link-decoration;
 
       &:not(:hover) {
         background-color: theme-color-level($color, 0);
+        border-color: theme-color-level($color, 0);
       }
     }
 
-    @include btn-disabled;
-
     &:not(:disabled):not(.disabled):active,
     &:not(:disabled):not(.disabled).active,
     .show > &.dropdown-toggle {
       background-color: theme-color-level($color, 1);
       box-shadow: 0 0 0 $btn-hover-width theme-color-level($color, 1);
+      border-color: theme-color-level($color, 1);
 
       &:focus,
       &.focus {
         box-shadow: 0 0 0 $input-btn-focus-width $box-shadow-button;
-        text-decoration: none;
+        text-decoration: $link-decoration;
 
         &:not(:hover) {
           background-color: theme-color-level($color, 0);
+          border-color: theme-color-level($color, 0);
         }
       }
     }
+
+    @include btn-disabled;
+  }
+  
+  .btn-outline-#{$color}{
+    color: theme-color-level($color, 0);
+
+    &:visited{
+      color: theme-color-level($color, 0);
+    }
+    
+    @include hover(){
+      color: $white;
+    };
+
+    &:focus,
+    &.focus {
+      color: color-yiq($value);
+    }
+
+    @if $color == "primary" {
+      color: $grisulado-900;
+      &:visited, &:hover {
+        color: $grisulado-900;
+      }
+    }
   }
 }
 
-.btn-link {
-  font-weight: $btn-font-weight;
+.btn-link, .btn-outline-link {
```

Comment:
Para no declarar la clase de esta manera, tanto `btn-link` & `btn-outline-link`, seria mejor y mas prolijo agregar una array dentro de **theme-colors** que incluya el color link.
Ahora teniendo eso, en el for donde estamos reccorriendo todos los colores, debemos usar el condicional if para modificar la clase link.
Tanto en `btn-#{color}` como `btn-outline-#{color}`

Reply:
```
$theme-colors: map-merge(
  $theme-colors,
  (
    'link': $ghost
  )
);
```